utils/src/server/rpc_module.rs

```diff
@@ -204,11 +206,12 @@ impl<Context> From<RpcModule<Context>> for Methods {
 
 impl<Context: Send + Sync + 'static> RpcModule<Context> {
 	/// Register a new synchronous RPC method, which computes the response with the given callback.
-	pub fn register_method<R, F>(&mut self, method_name: &'static str, callback: F) -> Result<(), Error>
+	pub fn register_method<R, F, E>(&mut self, method_name: &'static str, callback: F) -> Result<(), Error>
 	where
 		Context: Send + Sync + 'static,
 		R: Serialize,
-		F: Fn(RpcParams, &Context) -> Result<R, CallError> + Send + Sync + 'static,
+		F: Fn(RpcParams, &Context) -> Result<R, E> + Send + Sync + 'static,
+		E: Into<Error>,
```

Comment:
Now that we have three generics is perhaps a good time to use more explicit naming. I also think the more logical order would be `F, R, E` perhaps?
How about:
`F` –> `Method`,
`R` –> `RetVal`, 
`E` –> `MethodError`

Reply:
if we put `F, R, E`, you will probably be annoyed

Because `F` the closure that you provide is always known by the compiler so in practice you would have to do this:

`module.register_method::<_, (), Error>("hello_world", |_: RpcParams, _| Ok(())).unwrap();`

isn't `<R,E, F>` better then?

Reply:
I decided the revert the `Into<Error>` as this doesn't really work out-of-the-box when having the `client` in the API.

Basically, `Client::request -> Error` then we would require the custom error type to implement both `From<CustomError>` and `From<jsonrpsee::Error`, I prefer to tackle it another PR to actually make it work properly.